File: Programming EF Code First/chapter4/Chapter4/BAGA_FluentAPI/DataAccess/TripConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
  public class TripConfiguration : EntityTypeConfiguration<Trip>
  {
    public TripConfiguration()
    {
      HasKey(t => t.Identifier);

      Property(t => t.Identifier)
        .HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);

      Property(t => t.RowVersion)
        .IsRowVersion();
    }
  }
}

File: Programming EF Code First/chapter3/Chapter 3 VB/Fluent/DataAccess/TripConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model
Imports System.ComponentModel.DataAnnotations

Namespace DataAccess
  Public Class TripConfiguration
	  Inherits EntityTypeConfiguration(Of Trip)
	Public Sub New()
	  HasKey(Function(t) t.Identifier)

      Me.Property(Function(t) t.Identifier).HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity)

      Me.Property(Function(t) t.RowVersion).IsRowVersion()
	End Sub
  End Class
End Namespace
